repo/passwordless.js

```diff
@@ -0,0 +1,49 @@
+const error = require('error')
+const {db} = require('db')
+const {mapper} = require('repo/base')
+
+const map = mapper({
+  tokenRemote: 'token_remote',
+  tokenDirect: 'token_direct',
+  userId: 'user_id',
+  createdAt: 'created_at',
+})
+
+async function create (tokenRemote, tokenDirect, userId) {
+  return db.none(`
+    INSERT INTO
+    passwordless (token_remote, token_direct, user_id)
+    VALUES ($1, $2, $3)
+  `, [tokenRemote, tokenDirect, userId])
+  .catch({constraint: 'passwordless_pkey'}, error.db('passwordless.duplicate'))
+  .catch({constraint: 'passwordless_token_direct_key'}, error.db('passwordless.duplicate'))
+  .catch(error.db('db.write'))
+}
+
+async function getByTokens (tokenRemote, tokenDirect) {
+  return db.one(`
+    SELECT *
+    FROM passwordless
+    WHERE token_remote = $1
+    OR token_direct = $2
+  `, [tokenRemote, tokenDirect])
+  .then(map)
```

Comment:
Move this after catches.

Reply:
why?

Reply:
becouse if `map` throws, it is not a `db.read` error?

Reply:
map shouldn't throw. if it somehow does and that gets to prod, you still want that logged as a db.* error.

Reply:
Maps ideally shouldn't throw, but IMO, if it fails it is a lie to consider it a DB error.

Reply:
Or here `db.*` actually refers to repo errors?

Reply:
that's semantics. in practice `db.*` means don't crash the application on a db connection or wrong query error and make sure to log it on STDERR so we immediately get notified.

Reply:
Any internal error from routes will result in a 500 response and will be loged (even with LOG=0), with no crash. Also mapping errors are clearly not caused by db connection issues, and therefore to consider those related would be a misinformation, IMO.

Reply:
Also `.catch(...)` will not catch errors pre-read/write (like mapping to DB), so why consider post-read/write errors differently?